pkg/alertmanager/alertmanager_template.go

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package alertmanager
+
+import (
+	"encoding/json"
+	tmpltext "text/template"
+
+	tmplhtml "html/template"
+	"net/url"
+
+	"github.com/prometheus/alertmanager/template"
+)
+
+type grafanaExploreRange struct {
+	From string `json:"from"`
+	To   string `json:"to"`
+}
+
+type grafanaExploreQuery struct {
+	Datasource string `json:"datasource"`
+	Expr       string `json:"expr"`
+	Instant    bool   `json:"instant"`
+	Range      bool   `json:"range"`
+	RefID      string `json:"refId"`
+}
+
+type grafanaExploreParams struct {
+	Datasource string                `json:"datasource"`
+	Range      grafanaExploreRange   `json:"range"`
+	Queries    []grafanaExploreQuery `json:"queries"`
+}
+
+// grafanaURL is a template helper function to generate Grafana explore URL in the alert template.
+func grafanaURL(grafanaURL, datasource, from, to, expr string) string {
+	res, err := json.Marshal(&grafanaExploreParams{
+		Datasource: datasource,
+		Range: grafanaExploreRange{
+			From: from,
+			To:   to,
+		},
+		Queries: []grafanaExploreQuery{
+			{
+				Datasource: datasource,
+				Expr:       expr,
+				Instant:    false,
+				Range:      true,
+				RefID:      "A",
+			},
+		},
+	})
+	if err != nil {
+		panic(err)
```

Comment:
I'm a bit reluctant of including this especially for something that runs on a regular basis in the alertmanager. WDYT about returning a string which contains the error? It will result in ugly templates, but won't crash the process.

Reply:
Yeah, I'm not sure about that one. But the panic is wrong either way, the function should return error probably as described in https://pkg.go.dev/text/template#FuncMap

I do see your point on not failing, on the other way Alertmanager does expose metrics about failed template renders and if it does not fail, it might stay unnoticed otherwise you should have an alert on failed template renders anyway :shrug: 

[Alertmanager itself panics or returns errors but does not render the error as a result](https://github.com/prometheus/alertmanager/blob/main/template/template.go#L142), but I do not have a strong opinion, so if you prefer rendering it, I'll be happy to change it.

Reply:
For now, refactored to return the error properly in https://github.com/grafana/mimir/pull/3849/commits/57d203bd9a5f55b43f2f4b41da0743dbe526417b